src/main/java/meshIneBits/gui/MeshWindow.java

```diff
@@ -0,0 +1,713 @@
+        meshActionList.add(halfLengthBit);
+
+        MeshAction halfWidthBit = new MeshAction(
+                "halfWidthBit",
+                "Half Length",
+                "bit-half-width.png",
+                "Cut bit half in width",
+                "",
+                () -> meshController.scaleSelectedBit(
+                        100,
+                        50
+                ));
+        meshActionList.add(halfWidthBit);
+
+        MeshAction quartBit = new MeshAction(
+                "quartBit",
+                "1/4 Bit",
+                "bit-quart.png",
+                "Cut bit half in width and length",
+                "",
+                () -> meshController.scaleSelectedBit(
+                        50,
+                        50
+                ));
+        meshActionList.add(quartBit);
+
+        MeshAction deleteBit = new MeshAction(
+                "deleteBit",
+                "Delete Bit",
+                "bit-delete.png",
+                "Remove chosen bit(s)",
+                "DELETE",
+                meshController::deleteSelectedBits);
+        meshActionList.add(deleteBit);
+
+        MeshAction restoreBit = new MeshAction(
+                "restoreBit",
+                "Restore Bit",
+                "bit-restore.png",
+                "Restore to full bit",
+                "",
+                () -> meshController.scaleSelectedBit(
+                        100,
+                        100));
+        meshActionList.add(restoreBit);
+
+        MeshAction newBit = new MeshAction(
+                "newBit",
+                "New Bit",
+                "bit-new.png",
+                "Create new bit",
+                "alt N",
+                null) {
+            private UPPNewBit uppNewBit = new UPPNewBit(meshController);
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                if (meshController.getMesh() == null) {
+                    meshController.handleException(new Exception("Mesh not found"));
+                    return;
+                }
+                if (meshController.getCurrentLayer() == null) {
+                    meshController.handleException(new Exception("Layer not found"));
+                    return;
+                }
+                if (!meshController.getCurrentLayer().isPaved()) {
+                    meshController.handleException(new Exception("Layer not paved"));
+                    return;
+                }
+                toggleUtilityParametersPanel(uppNewBit);
+                meshController.setAddingBits(uppNewBit.isVisible());
+            }
+        };
+        meshActionList.add(newBit);
+
+        MeshToggleAction toggleShowSlice = new MeshToggleAction(
+                "toggleShowSlice",
+                "Show/Hide Layer Border",
+                "layer-border-toggle.png",
+                "Show or hide boundary of layer",
+                "",
+                meshController::toggleShowSlice
```

Comment:
Should be better to pass the "state" of the button than just toogle the state in the model. Issue occur during my test (button unpressed but action on)